blockmanager.go

```diff
@@ -1567,6 +1570,47 @@ func resolveFilterMismatchFromBlock(block *wire.MsgBlock,
 		return potentialBans, nil
 	}
 
+	// If all peers where serving filters consistent with the block, we
+	// cannot know for sure which one is dishonest (since we don't have the
+	// prevouts to deterministically reconstruct the filter). In this
```

Comment:
Not sure if I'm missing something, but since we have the full block, can't we just reconstruct the filter ourselves and check its hash? We can use ComputePkScript on each transaction input's sigScript/witness in order to retrieve the other output scripts needed. Might not always work if there's some type of non-standard script in the block though.

Reply:
Posting my answer that I think got lost in chat logs:

> we can compute the pkscript for supported types, and make sure at least they match in the filter

I tried this on recent blocks on mainnet, and we fail to interpret signature scripts containing uncompressed pubkeys as P2PKH
instead we think they are P2SH, and fail to match them with the filter
@wpaulino any reason for the ComputePKScript to assume only compressed pubkeys?
I also see that we are assuming signatures to be 71 or 72 bytes, while they can be other lengths, again causing us to assume p2sh

Reply:
No reason for it really. Would it help if those were to be checked for as well, or would we run into other unknown/unsupported scripts?

Reply:
I would think we never would be able to reconstruct all scripts, as someone could always make a non-standard one.

Reply:
Sure, I was more just wondering if it could be done for the majority of blocks if we address the `P2PKH` gaps.